src/extensions/common/hither/useHitherJob.ts

```diff
@@ -50,34 +50,24 @@ const useHitherJob = <T>(functionName: string, functionArgs: {[key: string]: any
         })
         return {result: undefined, job}
     }
-    else if (state.status === 'running') {
-        if (!state.job) throw Error('Unexpected: job is not defined')
-        return {
-            result: undefined,
-            job: state.job
-        }
-    }
-    else if (state.status === 'error') {
-        if (!state.job) throw Error('Unexpected: job is not defined')
-        return {
-            result: undefined,
-            job: state.job
-        }
+    // functionMatch() is true, i.e. the job is already being tracked. Return the record for it.
+    if (!state.job) throw Error('Unexpected: job is not defined')
+    let retrievedJob = {
+        result: undefined,
+        job: state.job
     }
-    else if (state.status === 'finished') {
-        if (!state.job) throw Error('Unexpected: job is not defined')
-        return {
-            result: state.job?.result,
-            job: state.job
-        }
-    }
-    else {
-        if (!state.job) throw Error('Unexpected: job is not defined')
-        return {
-            result: undefined,
-            job: state.job
-        }
+    switch (state.status) {
+        case 'finished':
+            retrievedJob.result = state.job?.result
+        // As of right now, for any other case we return the default set above.
+        // Anticipate this might be more customized in the future.
+        // eslint-disable-next-line no-fallthrough
+        case 'running':
+        case 'error':
+        default:
+            break;
```

Comment:
It turns out we only actually have two cases: either the job is finished (and we return the result), or it is not (and we leave the result undefined). I left it as a switch statement since I figured we might do more customization for specific statuses later, but if that assumption is incorrect we could make this even simpler.

Reply:
Looks like a good simplification to me.